src/tree_builder/actions.rs

```diff
@@ -747,7 +747,7 @@ impl<Handle, Sink> TreeBuilderActions<Handle>
     fn insert_element(&mut self, push: PushFlag, ns: Namespace, name: LocalName, attrs: Vec<Attribute>)
             -> Handle {
         declare_tag_set!(form_associatable =
-            "button" "fieldset" "input" "keygen" "label"
+            "button" "fieldset" "input" "keygen" "label" "legend"
```

Comment:
Hm, legend element's tests were not passing without this change. But maybe I missed something I'll look into that.